kb/prometheus-grafana-setup.md

```diff
@@ -0,0 +1,280 @@
+            .reportEvery(com.uber.m3.util.Duration.ofSeconds(10));
+
+  // for Prometheus collection, expose the scrape endpoint at port 8077. For example,
+  HttpServer scrapeEndpoint = MetricsUtils.startPrometheusScrapeEndpoint(registry, 8077);
+  // Stopping the starter will stop the http server that exposes the
+  // scrape endpoint.
+  //Runtime.getRuntime().addShutdownHook(new Thread(() -> scrapeEndpoint.stop(1)));
+
+  //Create WF service stubs to connect to the front end service.
+  WorkflowServiceStubs service = WorkflowServiceStubs.newServiceStubs(
+         WorkflowServiceStubsOptions.newBuilder()
+                 .setMetricsScope(scope) //set the metrics scope for the WorkflowServiceStubs
+                 .build());
+
+  //Create a Workflow service client which can be used to start, Signal, and Query Workflow Executions.
+  WorkflowClient yourClient = WorkflowClient.newInstance(service,
+         WorkflowClientOptions.newBuilder().build());
+
+  //...
+```
+
+You can set up separate scrape endpoints in your Clients that you use to start your Workers and Workflow Executions.
+To use this example, add this example code with the Prometheus endpoint on port 8077 in your Worker program, and use `yourClient` to start your Workers.
+Similarly, in your starter code, add this example code and set the Prometheus endpoint to port 8078, create a Workflow Client (as described in the code example), and use it to start your Workflow Execution.
+
+For more examples on how this is set across SDKs, see the metrics samples:
+
+- [Java SDK Samples](https://github.com/temporalio/samples-java/tree/main/src/main/java/io/temporal/samples/metrics)
+- [Go SDK Samples](https://github.com/temporalio/samples-go/tree/main/metrics)
+
+In your Workers, you can set specific WorkerOptions for performance tuning, as described in the [Worker Performance Guide](/application-development/worker-performance).
+With the scrape endpoints set, define your Prometheus scrape configuration and targets to receive the metrics data from the Temporal Cluster and Temporal SDKs.
+
+### Prometheus configuration
+
+Enable Prometheus to scrape the metrics from the endpoints defined in the Cluster and SDK configurations.
+
+For example with the local docker-compose Temporal Cluster, you can create a separate container for Prometheus:
+
+```
+version: "3.5"
+services:
+#...
+  prometheus:
+   container_name: prometheus
+   image: prom/prometheus:v2.37.0
+   ports:
+     - 9090:9090
+   volumes:
+     - type: bind
+       source:./deployment/prometheus/config.yml
+       target: /etc/prometheus/prometheus.yml
+   depends_on:
+     - temporal
+#...
+```
+
+In this example, we use a Prometheus docker image for v2.37.0 and the default ports for the container, and set the path to a Prometheus configuration file for the docker container. For this example, create a Prometheus config.yml file at ./deployment/prometheus in your docker-compose Temporal Cluster project.
```

Comment:
Re: version of prometheus -
Maybe link to where the current Prometheus docker build image is listed: https://hub.docker.com/r/prom/prometheus